test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs

```diff
@@ -565,6 +601,52 @@ public void EnumerableToCreatedQueue()
     }
 
     [Fact]
+    public void ArrayToCreatedStack()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public int[] Value { get; } }",
+            "class B { public Stack<long> Value { get; set; } }");
+        TestHelper.GenerateMapper(source)
+            .Should()
+            .HaveMethodBody("MapToStack",
+                """
+                var target = new System.Collections.Generic.Stack<long>();
+                target.EnsureCapacity(source.Length + target.Count);
```

Comment:
This is not yet optimal, the built in collections all have a ctor accepting a capacity parameter which would reduce the array allocations and could be used by Mapperly. However, this could also be adressed in a separate PR.

Reply:
> However, this could also be adressed in a separate PR.

I'd planned on updating the issue after completing this PR. I was going to look at how you did this with dictionaries.

What concrete types is this needed for? Types like `Stack<T>`, `Queue<T>` have constructors that take `IEnumerable<T>`  so I was going to use that where possible.

Reply:
Sounds good to me. `Stack<T>`, `Queue<T>` are all, as I think `List<T>` and co are always mapped to via linq.

Reply:
Should I make those changes a separate PR?

Reply:
I'd do it separately if possible.